### DISCUSSION TITLE
How to control the accuracy of solvers?

### DISCUSSION THREAD

**Discussion by erelsgl** [cvxpy/cvxpy#1782] [Q&A]
I tried to solve a simple convex optimization problem using the ECOS solver. I know that the solution should be (0, 50, 50, 100). The solution I received was: (8.683552667382458e-08, 50.00000392080848, 50.000003920808496, 99.99999189292947). It is quite accurate, but I wanted a more accurate solution, so I tried to use the [parameters to ECOS](https://www.cvxpy.org/tutorial/advanced/index.html), e.g.:

      problem.solve(solver=cvxpy.ECOS, abstol=1e-10)

I tried all other parameters: reltol, feastol, abstol_inacc, etc. I also tried to set max_iters=100000. In all cases, I got exactly the same result - as if the paramter has no effect.

What is the right way to increase the accuracy of a solver?

**Comment by rileyjmurray** [comment]
Sadly it is not really possible to compute solutions to more than 8 digits
of accuracy. The reason is that convex optimization algorithms need to
multiply the primal variables and the dual variables to check optimality.
That multiplication is only accurate up to double precision unit round-off,
which is about 1e-16. Since the product of numbers on the order of 1e-8 is
1e-16, you can’t reliably compute the optimization variables to error less
than 1e-8.

**Comment by erelsgl** [comment]
Is there a way to use `long double`, or a specialized type that provides an even higher accuracy, to improve the solver accuracy?

**Comment by rileyjmurray** [comment]
You would need to modify the ECOS source code to do that. As of this moment
none of CVXPY’s solvers support precision higher (or lower) than double.

**Comment by huangjiancong1** [comment]
> I tried to solve a simple convex optimization problem using the ECOS solver. I know that the solution should be (0, 50, 50, 100). The solution I received was: (8.683552667382458e-08, 50.00000392080848, 50.000003920808496, 99.99999189292947). It is quite accurate, but I wanted a more accurate solution, so I tried to use the [parameters to ECOS](https://www.cvxpy.org/tutorial/advanced/index.html), e.g.:
> 
> ```
>   problem.solve(solver=cvxpy.ECOS, abstol=1e-10)
> ```
> 
> I tried all other parameters: reltol, feastol, abstol_inacc, etc. I also tried to set max_iters=100000. In all cases, I got exactly the same result - as if the paramter has no effect.
> 
> What is the right way to increase the accuracy of a solver?

have you improved the accuracy?

**Reply by erelsgl** [reply]
No... I had to change the algorithm to accomodate for the lower accuracy.
